src/panels/config/js/trigger/geo_location.js

```diff
@@ -0,0 +1,117 @@
+import { h, Component } from "preact";
```

Comment:
rename file to `geo_location.ts`

Reply:
I just tried that, but that apparently has all sorts of implications on the structure of the code. In particular, the modifications required to the `render` method didn't seem to result in a proper trigger form. Do you maybe have an example of how to transform an `extends Component` into TypeScript?

Reply:
Because we haven't done it before, I'll let it slip this time